metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -147,9 +147,8 @@ final class RenameProvider(
     fileChanges
       .find { file =>
         isOccurence(str => {
-          (str.desc.isType || str.desc.isTerm) && file.endsWith(
-            str.desc.name.value + ".scala"
-          )
+          (str.desc.isType || str.desc.isTerm) &&
+            Paths.get(new URI(file)).filename == str.desc.name.value + ".scala"
```

Comment:
Since we know `file` is a URI, can we do this instead?
```suggestion
           file.endsWith(s"/${str.desc.name.value}.scala")
```

Reply:
That's what @tgodzik suggested too. If you both think it's a good idea, I'll fix it this way!